src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -107,16 +124,21 @@ public static IServiceCollection AddAsyncInitializer(this IServiceCollection ser
 
         private class DelegateAsyncInitializer : IAsyncInitializer
         {
-            private readonly Func<Task> _initializer;
+            private readonly Func<CancellationToken, Task> _initializer;
 
-            public DelegateAsyncInitializer(Func<Task> initializer)
+            public DelegateAsyncInitializer(Func<CancellationToken, Task> initializer)
             {
                 _initializer = initializer;
             }
 
-            public Task InitializeAsync()
+
+            ///<inheritdoc/>
+            public Task InitializeAsync() => InitializeAsync(CancellationToken.None);
```

Comment:
Why add this method? The class is private, so even if the method is public, it will never be exposed, and it's not called by anything anyway.